### ISSUE TITLE
Liquibase could not start correctly, Reason: liquibase.exception.DatabaseException: Syntax error in SQL statement

### ISSUE THREAD

**Issue by rjbgaspar** [jhipster/generator-jhipster#16667]
##### **Overview of the issue**

Create JHipster project from a JDL file. 
Run `.\mvnw` command and all liquibase migrations fail
```
2021-10-13 16:50:57.610 ERROR 54784 --- [ panther-task-1] t.j.c.liquibase.AsyncSpringLiquibase     : Liquibase could not start correctly, your database is NOT ready: liquibase.exception.MigrationFailedException: Migration failed for change set config/liquibase/changelog/20211013101326_added_entity_CadErpBinding.xml::20211013101326-1-data::jhipster:
     Reason: liquibase.exception.DatabaseException: Syntax error in SQL statement "IF[*] EXISTS(SELECT TABLE_NAME
            FROM INFORMATION_SCHEMA.COLUMNS
           WHERE TABLE_NAME = 'cad_erp_bindings'
             AND TABLE_SCHEMA = 'PUBLIC'
             AND COLUMNPROPERTY(OBJECT_ID(TABLE_SCHEMA + '.' + TABLE_NAME), COLUMN_NAME, 'IsIdentity') = 1)
        SET IDENTITY_INSERT PUBLIC.CAD_ERP_BINDINGS ON"; SQL statement:
IF EXISTS(select TABLE_NAME
            from INFORMATION_SCHEMA.COLUMNS
           where TABLE_NAME = 'cad_erp_bindings'
             and TABLE_SCHEMA = 'PUBLIC'
             and COLUMNPROPERTY(object_id(TABLE_SCHEMA + '.' + TABLE_NAME), COLUMN_NAME, 'IsIdentity') = 1)
        SET IDENTITY_INSERT PUBLIC.cad_erp_bindings ON [42000-200] [Failed SQL: (42000) IF EXISTS(select TABLE_NAME
            from INFORMATION_SCHEMA.COLUMNS
           where TABLE_NAME = 'cad_erp_bindings'
             and TABLE_SCHEMA = 'PUBLIC'
             and COLUMNPROPERTY(object_id(TABLE_SCHEMA + '.' + TABLE_NAME), COLUMN_NAME, 'IsIdentity') = 1)
        SET IDENTITY_INSERT PUBLIC.cad_erp_bindings ON]
```

##### **Motivation for or Use Case**

This issue is related with JHipster version 7.2.0 and 7.3.0 since the same jdl works fine is 7.1.0

##### **Reproduce the error**

Run jhipster jdl entites.jdl with database type SQL and dev database type set to h2Disk

##### **Related issues**
Not found

##### **Suggest a Fix**
Not applicable

INFO! Using JHipster version installed locally in current project's node_modules
##### **JHipster Version(s)**

```
panther@0.0.0 C:\Users\Ricardo Gaspar\code\wasp-test\panther
`-- generator-jhipster@7.3.0

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "authenticationType": "oauth2",
    "cacheProvider": "no",
    "serverPort": "50081",
    "serviceDiscoveryType": "eureka",
    "skipUserManagement": true,
    "baseName": "panther",
    "buildTool": "maven",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "enableHibernateCache": false,
    "enableSwaggerCodegen": false,
    "enableTranslation": true,
    "jhiPrefix": "jhi",
    "languages": ["en"],
    "messageBroker": false,
    "prodDatabaseType": "mssql",
    "searchEngine": false,
    "skipClient": true,
    "testFrameworks": [],
    "websocket": false,
    "enableGradleEnterprise": false,
    "gradleEnterpriseHost": "",
    "reactive": true,
    "packageName": "com.gv.panther",
    "packageFolder": "com/gv/panther",
    "applicationType": "microservice",
    "jhipsterVersion": "7.3.0",
    "clientPackageManager": "npm",
    "dtoSuffix": "DTO",
    "entitySuffix": "",
    "applicationIndex": 1,
    "skipServer": false,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "blueprints": [],
    "otherModules": [],
    "pages": [],
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "withAdminUi": false,
    "nativeLanguage": "en",
    "creationTimestamp": 1634119946658,
    "jwtSecretKey": "YourJWTSecretKeyWasReplacedByThisMeaninglessTextByTheJHipsterInfoCommandForObviousSecurityReasons",
    "entities": [
      "CadErpBinding",
      "Config",
      "Counter",
      "ImosOrganizerProcess",
      "ImosOrganizerProcessError",
      "ImosOrganizerProcessOrder",
      "ImosOrganizerProcessPart",
      "StefaniEvolutionOneMechanizationPreset",
      "WorkCadBinding"
    ],
    "lastLiquibaseTimestamp": 1634120486000
  }
}

</pre>
</details>

##### **JDL for the Entity configuration(s) `entityName.json` files generated in the `.jhipster` directory**

<details>
<summary>JDL entity definitions</summary>

<pre>
entity CadErpBinding (cad_erp_bindings) {
  type String maxlength(20)
  typology String maxlength(50)
  description String maxlength(50)
  comments String
  erpBinding String
  cadBinding String
  createdAt Instant
  updatedAt Instant
  colore String maxlength(255)
  dz String maxlength(255)
  rifx String maxlength(255)
  rify String maxlength(255)
  rifcodax String maxlength(255)
  rifcoday String maxlength(255)
  vena String maxlength(255)
  nascosto String maxlength(255)
  profilo String maxlength(255)
  veltaglio String maxlength(255)
  velrotaz String maxlength(255)
  minrestox String maxlength(255)
  minrestoy String maxlength(255)
  parresti String maxlength(255)
  veltaglioy String maxlength(255)
  velrotazy String maxlength(255)
  velapprocy String maxlength(255)
  velapprocx String maxlength(255)
}
entity Config (configs) {
  slug String required unique
  description String
  strategy String
  value String
  createdAt Instant
  updatedAt Instant
}
entity Counter (counters) {
  document String
  value Integer
  createdAt Instant
  updatedAt Instant
  series String
}
entity ImosOrganizerProcess (imos_organizer_processes) {
  userName String
  complete Boolean
  fileName String
  fileHeader String
  createdAt Instant
  updatedAt Instant
  infoMessagesCount Integer
  warningMessagesCount Integer
  errorMessagesCount Integer
  criticalMessagesCount Integer
  failedAt Instant
  publishesCount Integer
  downloadsCount Integer
}
entity ImosOrganizerProcessError (imos_organizer_process_errors) {
  errorLevel String
  errorMessage String
  errorSource String
  createdAt Instant
  updatedAt Instant
}
entity ImosOrganizerProcessOrder (imos_organizer_process_orders) {
  orderName String
  createdAt Instant
  updatedAt Instant
  revisionNumber Integer
}
entity ImosOrganizerProcessPart (imos_organizer_process_parts) {
  orderId Integer
  orderName String
  articleId Integer
  articleName String
  partId String
  camCncFilePath String
  multiJobLine String
  createdAt Instant
  updatedAt Instant
  partName String
  length BigDecimal
  width BigDecimal
  thickness BigDecimal
  color String
  quantity Integer
  overQuantity Integer
  grain Integer
  partCode String
  partDescription String
  typologie String
  part Integer
  idMag Integer
  indicePrg Integer
  indiceOpt Integer
  otimize Integer
  priority Integer
  all1 Integer
  aff1 Integer
  exitWorkCenter Integer
  compose Integer
  scaricoRuotato Integer
  stackHeight BigDecimal
  veneer Integer
  edge1 String
  edge2 String
  edge3 String
  edge4 String
  sovraMatX BigDecimal
  sovraMatY BigDecimal
  futureUse1 Integer
  futureUse2 Integer
  futureUse3 BigDecimal
  futureUse4 BigDecimal
  note String
  numberOfTextFields Integer
  textField1 String
  textField2 String
  textField3 String
  textField4 String
  textField5 String
  textField6 String
  textField7 String
  textField8 String
  textField9 String
  textField10 String
  textField11 String
  textField12 String
  textField13 String
  textField14 String
  textField15 String
  textField16 String
  textField17 String
  textField18 String
  textField19 String
  textField20 String
  textField21 String
  textField22 String
  textField23 String
  textField24 String
  textField25 String
  textField26 String
  textField27 String
  textField28 String
  textField29 String
  textField30 String
  textField31 String
  textField32 String
  textField33 String
  textField34 String
  textField35 String
  textField36 String
  textField37 String
  textField38 String
  textField39 String
  textField40 String
  sortKey Integer
  articlePosition String
}
entity StefaniEvolutionOneMechanizationPreset (stefani_evolution_one_mechanization_presets) {
  description String required
  comments String required
  programNumber Integer required
  slot Integer required
  toupieU String required
  a BigDecimal
  b BigDecimal
  toolDiameter BigDecimal
  createdAt Instant
  updatedAt Instant
  start BigDecimal
  end BigDecimal
  type String
  machine String
}
entity WorkCadBinding (work_cad_bindings) {
  imosOrganizerProcessId Long required
  typology String
  typologyTracking String
  typologyCount Integer
  materialId String maxlength(32)
  materialGrainId String maxlength(3)
  materialGrainAngle BigDecimal
  materialCategory String maxlength(80)
  materialOrderId String maxlength(32)
  materialRender String maxlength(32)
  materialThickness BigDecimal
  surfaceTopId String maxlength(32)
  surfaceTopGrainId String maxlength(3)
  surfaceTopGrainAngle BigDecimal
  colorTop String maxlength(80)
  surfaceBottomId String maxlength(32)
  surfaceBottomGrainId String maxlength(3)
  surfaceBottomGrainAngle BigDecimal
  colorBottom String maxlength(80)
  surfaceLoc Integer
  createdAt Instant
  updatedAt Instant
}
relationship ManyToOne {
  ImosOrganizerProcessError{imosOrganizerProcessPart required} to ImosOrganizerProcessPart
  ImosOrganizerProcessOrder{imosOrganizerProcess(belongsTo) required} to ImosOrganizerProcess
  ImosOrganizerProcessPart{imosOrganizerProcess required} to ImosOrganizerProcess
  ImosOrganizerProcessPart{imosOrganizerProcessOrder required} to ImosOrganizerProcessOrder
}

microservice CadErpBinding, Config, Counter, ImosOrganizerProcess, ImosOrganizerProcessError, ImosOrganizerProcessOrder, ImosOrganizerProcessPart, StefaniEvolutionOneMechanizationPreset, WorkCadBinding with panther
clientRootFolder CadErpBinding, Config, Counter, ImosOrganizerProcess, ImosOrganizerProcessError, ImosOrganizerProcessOrder, ImosOrganizerProcessPart, StefaniEvolutionOneMechanizationPreset, WorkCadBinding with panther

</pre>
</details>

##### **Environment and Tools**

java version "15.0.1" 2020-10-20
Java(TM) SE Runtime Environment (build 15.0.1+9-18)
Java HotSpot(TM) 64-Bit Server VM (build 15.0.1+9-18, mixed mode, sharing)

git version 2.23.0.windows.1

node: v14.17.6

npm: 7.20.5

Docker version 20.10.8, build 3967b7d

docker-compose version 1.29.2, build 5becea4c

Congratulations, JHipster execution is complete!
Sponsored with ÔØñ´©Å  by @oktadev.

**Comment by rjbgaspar** [comment]
I just checked with JHipster 7.4.1 and problem still persist.

**Comment by DanielFran** [comment]
This has been fixed in 7.5.0: https://github.com/jhipster/generator-jhipster/pull/16612.
